File: Java/Numbers/UnitConverter/Utilities.java
/**
 * Created by sraone on 10/5/14.
 */

import java.util.ArrayList;
import java.util.Scanner;

public class Utilities {

    private static Scanner S = new Scanner(System.in );

    public static float get_value () {
        while (true) {
            try {
                float input = Float.parseFloat(S.nextLine());
                return input;
            } catch (Exception e) {
                System.out.print("Please enter a valid float value ... ");
            }
        }
    }

    public static String get_metric (ArrayList<String> metrics) {
        System.out.print("Please enter the Metric ..." + metrics + " ... ");
//        System.out.println(metrics);
        while (true) {
            String metric = S.nextLine().toUpperCase();
            if (metrics.contains(metric)) return metric;
            else System.out.print("Sorry I cannot find the metric you want\nPlease try again ... ");
        }
    }

    public static String get_conversion (String metric, ArrayList<String> metrics) {
        System.out.println("Conversions available ... ");
        for (String m : metrics ){
            if (!(m.equalsIgnoreCase(metric))) System.out.println(metric + " --> " + m);
        }
        System.out.print("\nPlease select target metric ... ");
        while(true){
            String target = S.nextLine().toUpperCase();
            if ( (! (target.equals(metric) )) && metrics.contains(target)) return target;
            else System.out.println("Please enter the metric displayed in the above menu ... \nPlease try again ... ");
        }
    }
}

File: Python/Numbers/UnitConverter/utilities.py
'''
Created on 11-Oct-2014

@author: ghantasa
'''
def get_input (unit, metrics):
    value = float(input('Please enter the ' + unit + ' ... '))
    while (True):
        metric = input ('Please select the metric [ ' + ','.join(metrics) + ' ] ... ').upper()
        if metric in metrics : break
        else : print('There is not such metric! ')
    
    while (True):
        print_conversions(metrics, metric)
        target = input ('Please enter a target metric for conversion ... ').upper()
        if target in metrics : break
        else : print('There is no such metric! ')
        
    return  (value, metric, target)
     
def print_conversions (metrics, current_metric):
    for i in metrics:
        if not i == current_metric.upper() : print ("\t" + current_metric.upper() + ' --> ' + i)
        
def print_output (value, metric, result, target):
    print (str(value) +" " + metric + " is equal to " + str(result) + ' ' + target + '\n')